lib/httpi/auth/ssl.rb

```diff
@@ -10,7 +10,15 @@ class SSL
 
       VERIFY_MODES = [:none, :peer, :fail_if_no_peer_cert, :client_once]
       CERT_TYPES = [:pem, :der]
-      SSL_VERSIONS = OpenSSL::SSL::SSLContext::METHODS.reject { |method| method.match(/server|client/) }.sort.reverse
+
+      # Fix for
+      # httpi/auth/ssl.rb:13: warning: constant OpenSSL::SSL::SSLContext::METHODS is deprecated
+      ssl_context = OpenSSL::SSL::SSLContext
+      SSL_VERSIONS = if ssl_context.const_defined? :METHODS_MAP
+        ssl_context.const_get(:METHODS_MAP).keys
+      else
+        ssl_context::METHODS.reject { |method| method.match(/server|client/) }
```

Comment:
Why are you using `const_get` in one case and `::` in other?

Reply:
because METHODS_MAP is a private constant.

Reply:
Because METHODS_MAP is marked as private
Look at: https://github.com/ruby/ruby/blob/trunk/ext/openssl/lib/openssl/ssl.rb#L222

Reply:
Well, it sucks that we are using private constants from other libs :(

Reply:
> Well, it sucks that we are using private constants from other libs :(

@tycooon I agree with you.

Someone suggests a better approach? 

These values change by different `OpenSSL::SSL::SSLContext`? We can't set this values on `SSL_VERSIONS` directly?

Reply:
I only see either fetching it from the OpenSSL lib as @webhoernchen is doing it or defining it directly `SSL_VERSIONS = [:TLSv1_2, :TLSv1_1, :TLSv1, :SSLv3, :SSLv23]` but having to update it manually if anything changes.

Reply:
@webhoernchen what do you think? I see no problem with `SSL_VERSIONS = [:TLSv1_2, :TLSv1_1, :TLSv1, :SSLv3, :SSLv23]` approach.